File: learn_ruby/02_calculator/calculator.rb
def add (total, addend)
	total + addend
end

def subtract (minuend, subtrahend)
	minuend - subtrahend
end

def sum (series)
	series.reduce(0) {|total, addend| add(total, addend)}
end

def multiply (*args)
	args.reduce(1) {|multiplicand, multiplier| multiplicand * multiplier}
end

def power (base, exponent)
	base ** exponent
end

def factorial (number)
	return 0 if (number == 0)

	total = 1
	number.times do |i|
		total *= (i + 1)
	end

	total
end

File: web_development_101/javascript-exercises/calculator/calculator.js
function add (addendA, addendB) {
	return addendA + addendB;
}

function subtract (minuend, subtrahend) {
	return minuend - subtrahend;
}

function sum (addends) {
	let total = 0;
	addends.forEach(addend => total += addend);
	return total;
}

function multiply (multipliers) {
	if (!multipliers[0])
		return;	
	let product = 1;
	multipliers.forEach(multiplier => product *= multiplier);
	return product;
}

function power(base, exponent) {
	if(exponent === 0)
		return 1;
	return base * power(base, --exponent);
}

function factorial(value) {
	if(value === 0)
		return 1;
	return value * factorial(--value);
}

module.exports = {
	add,
	subtract,
	sum,
	multiply,
    power,
	factorial
}